app/models/book.rb

```diff
@@ -1,3 +1,11 @@
 class Book < ApplicationRecord
-  
+  validates :title, presence: true
+
+  has_many :reviews
+
+  def self.average_ratings
+    select("books.*, AVG(rating) AS avg_rating")
+      .joins(:reviews)
+      .group(:id)
+  end
```

Comment:
Building off the comment in the controller:
From an instance method you could call `reviews.average(:rating)` where you utilize the relationship between book and reviews. You'd use a similar strategy using `maximum(:rating)` and `minimum(:rating)`